plugin/storage/memory/memory.go

```diff
@@ -71,6 +71,20 @@ func (m *Store) GetDependencies(ctx context.Context, endTs time.Time, lookback t
 		trace, _ := m.deduper.Adjust(orig)
 		if m.traceIsBetweenStartAndEnd(startTs, endTs, trace) {
 			for _, s := range trace.Spans {
+				// this finds unmanged peers of the service. So this are all external dependcies
+				peer := m.findPeer(s)
```

Comment:
This will not address the issue in the ticket. While peer.service is a recommended tag, it's often impossible for the instrumentation to know.

Reply:
How you mean? What is the case this tag doesn't catch? And shouldn't this be more an issue for the collector? Shouldn't the observability layer trust the open telemetry spec?

Reply:
Or did you referring too the other informations that the open telemetry gives for peers?
Something like

- peer.port
- peer.ipv6
- peer.ipv4
- peer.hostname
- peer.address

So we should combine all these information for the key?

Reply:
There are two issues.
  1. You're force-adding a dependency based on the presence of `peer.service` tag regardless of whether the current node is the leaf in the call graph or an inner node that would produce a dependency anyway from its child span.
  2. None of the tags you listed are actually _required_ to be present in the span. In a typical HTTP client instrumentation, the instrumentation layer does not know the name of the downstream service, only its URL, so `peer.service` tag is likely to be missing, while other `peer.*` tags might be present. We would need some heuristic function that would decide how to name the downstream service in this case.